lib/lackeys/registry.rb

```diff
@@ -139,17 +146,39 @@ def method?(method_name)
     def call(method_name, *args, &block)
       method_name = method_name.to_sym
       raise "#{method_name} has not been registered" unless method? method_name
-      return_values = []
+      return_values = {}
       commit_chain = []
+      is_multi = value_hash[method_name][:multi]
+
       value_hash[method_name][:observers].each do |obs|
         cached_obs = observer_cache.fetch(obs)
-        return_values << cached_obs.send(method_name, *args, &block)
-        commit_chain << cached_obs
+        res = cached_obs.send(method_name, *args, &block)
+        if is_multi
```

Comment:
maybe pass this off to a private method that processes what to return (array or hash)